Orange/classification/tree.py

```diff
@@ -64,6 +64,8 @@ def __init__(
         self.min_samples_split = min_samples_split
         self.sufficient_majority = sufficient_majority
         self.max_depth = max_depth
+        self.params = {k: v for k, v in vars().items()
+                       if k not in ('args', 'kwargs')}
```

Comment:
You're right, it's uglier (although perhaps better).

Leave it like this for now, and perhaps change this in all SKL learners. Perhaps by adding something like this to `SKLLearner`

    from inspect import signature

    ...

    def update_params(self, values):
        param_names = signature(type(self).__init__).parameters
        self.params.update({name: values[name] for name in param_names if name not in {"args", "kwargs"})

which would then be called from `__init__` of derived classes with `self.update_params(vars())`. This would only copy the names that were given as arguments.

I see that there's already some similar magic in our SKL wrappers.

Not in this PR.